src/Codebase/Infrastructure/Service/CodebaseService.php

```diff
@@ -19,12 +23,42 @@ class CodebaseService implements CodebaseServiceInterface
      */
     protected $sourceParser;
 
+    /**
+     * @var \Codebase\Infrastructure\ProjectConfigurationParser\ProjectConfigurationParserInterface
+     */
+    protected ProjectConfigurationParserInterface $projectConfigurationParser;
+
     /**
      * @param \Codebase\Infrastructure\SourceParser\SourceParser $sourceParser
+     * @param \Codebase\Infrastructure\ProjectConfigurationParser\ProjectConfigurationParserInterface $projectConfigurationParser
      */
-    public function __construct(SourceParser $sourceParser)
+    public function __construct(SourceParser $sourceParser, ProjectConfigurationParserInterface $projectConfigurationParser)
     {
         $this->sourceParser = $sourceParser;
+        $this->projectConfigurationParser = $projectConfigurationParser;
+    }
+
+    /**
+     * @param \Evaluate\Infrastructure\Configuration\ConfigurationProvider $configurationProvider
+     *
+     * @return \Codebase\Application\Dto\CodebaseSourceDto
+     */
+    public function parse(ConfigurationProvider $configurationProvider): CodebaseSourceDto
+    {
+        $projectConfigurationRequest = new ConfigurationRequestDto(
+            $configurationProvider->getToolingConfiguration(),
+            $configurationProvider->getSrcDirectory(),
+        );
+        $projectConfiguration = $this->parseProjectConfiguration($projectConfigurationRequest);
```

Comment:
Please remove the current logic from the service layer.

Reply:
Thanks, the logic moved to CodeBaseReader